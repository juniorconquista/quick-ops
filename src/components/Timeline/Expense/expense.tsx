import React, { memo, useMemo } from 'react';
import { Box, H2, H1, P } from 'quarks-ui';
import { css } from 'styled-components';

import { Card } from '../../../containers/Timeline/timeline.types';
import { formatDate, handleIcon } from '../timeline.utils';
import Icon from '../Icon';
import Status from '../Status';

const Expense: React.FC<Card & any> = (props) => {
    const ExpenseIcon = useMemo(() => handleIcon(props.expenseTypeIcon), [
        props.expenseTypeIcon,
    ]);
    return (
        <Box
            styling="row"
            bg="white.default"
            width="100%"
            mb="24px"
            minHeight="154px"
            p={{ mobile: '20px', tablet: '20px 30px', desktop: '20px 40px' }}
            alignItems="stretch"
            justifyContent="space-between"
        >
            <Box
                styling="row"
                width="70%"
                css={css({
                    '@media (max-width: 1250px)': {
                        width: '80%',
                    },
                })}
                flexDirection={{
                    mobile: 'column',
                    tablet: 'row',
                    desktop: 'row',
                }}
                justifyContent="flex-start"
                alignItems={{
                    mobile: 'flex-start',
                    tablet: 'center',
                    desktop: 'center',
                }}
            >
                <Box
                    styling="column"
                    mr={{ mobile: '0px', tablet: '40px', desktop: '40px' }}
                    mb={{ mobile: '15px', tablet: '15px', desktop: '0px' }}
                    flexDirection={{
                        row: 'row',
                        tablet: 'column',
                        desktop: 'column',
                    }}
                >
                    <Icon styling={props?.expenseTypeCode ?? 'base'}>
                        <ExpenseIcon />
                    </Icon>
                    <H2
                        styling="base"
                        fontSize="13px"
                        ml={{ mobile: '5px', tablet: '0px', desktop: '0px' }}
                    >
                        {formatDate(props.cardDate)}
                    </H2>
                </Box>
                <Box
                    styling="row"
                    justifyContent={{
                        mobile: 'flex-start',
                        tablet: 'space-between',
                        desktop: 'space-between',
                    }}
                    flexDirection={{
                        mobile: 'column',
                        tablet: 'row',
                        desktop: 'row',
                    }}
                    alignItems={{
                        mobile: 'flex-start',
                        tablet: 'center',
                        desktop: 'center',
                    }}
                    width="100%"
                >
                    <Box
                        styling="column"
                        alignItems="flex-start"
                        mr={{ mobile: '0px', tablet: '0px', desktop: '15px' }}
                        mb={{ mobile: '10px', tablet: '0px', desktop: '0px' }}
                        flexDirection={{
                            row: 'row',
                            tablet: 'column',
                            desktop: 'column',
                        }}
                    >
                        <H1
                            styling="base"
                            fontWeight="bold"
                            fontSize={{
                                mobile: '14px',
                                tablet: '11px',
                                desktop: '11px',
                            }}
                            mb={{
                                mobile: '0px',
                                tablet: '10px',
                                desktop: '10px',
                            }}
                            mr={{
                                mobile: '5px',
                                tablet: '0px',
                                desktop: '0px',
                            }}
                        >
                            TIPO
                        </H1>
                        <H2
                            styling="base"
                            fontWeight="bold"
                            color="text.contrast"
                            fontSize={{
                                mobile: '14px',
                                tablet: '20x',
                                desktop: '20x',
                            }}
                            mb={{
                                mobile: '0px',
                                tablet: '10px',
                                desktop: '10px',
                            }}
                            mr={{
                                mobile: '5px',
                                tablet: '0px',
                                desktop: '0px',
                            }}
                        >
                            {props.expenseTypeCode}
                        </H2>
                        <P styling="base" fontSize="14px">
                            {props.notes}
                        </P>
                    </Box>
                    <Box
                        styling="column"
                        alignItems="flex-start"
                        mb={{ mobile: '10px', tablet: '0px', desktop: '0px' }}
                        flexDirection={{
                            row: 'row',
                            tablet: 'column',
                            desktop: 'column',
                        }}
                    >
                        <H1
                            styling="base"
                            fontWeight="bold"
                            fontSize={{
                                mobile: '14px',
                                tablet: '11px',
                                desktop: '11px',
                            }}
                            mb={{
                                mobile: '0px',
                                tablet: '10px',
                                desktop: '10px',
                            }}
                            mr={{
                                mobile: '5px',
                                tablet: '0px',
                                desktop: '0px',
                            }}
                        >
                            VALOR
                        </H1>
                        <H2
                            styling="base"
                            fontWeight="bold"
                            color="text.contrast"
                            fontSize={{
                                mobile: '14px',
                                tablet: '20x',
                                desktop: '20x',
                            }}
                            mb={{
                                mobile: '0px',
                                tablet: '10px',
                                desktop: '10px',
                            }}
                            mr={{
                                mobile: '5px',
                                tablet: '0px',
                                desktop: '0px',
                            }}
                        >
                            {`${props.currencySymbol} ${props.amountSpent}`}
                        </H2>
                        <P styling="base" fontSize="14px">
                            {`Valor da nota: ${props.currencySymbol} ${props.amountTotal}`}
                        </P>
                    </Box>
                    <Box
                        styling="column"
                        alignItems="flex-start"
                        flexDirection={{
                            row: 'row',
                            tablet: 'column',
                            desktop: 'column',
                        }}
                    >
                        <H1
                            styling="base"
                            fontSize="11px"
                            fontWeight="bold"
                            mb="10px"
                        >
                            STATUS
                        </H1>
                        {props?.status && (
                            <Status styling={props?.status ?? 'base'}>
                                {props?.status}
                            </Status>
                        )}
                    </Box>
                </Box>
            </Box>
            <Box styling="row">
                <P> Ver nota fiscal</P>
            </Box>
        </Box>
    );
};

export default memo(Expense);
