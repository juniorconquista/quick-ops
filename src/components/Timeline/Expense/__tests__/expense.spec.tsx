import React from 'react';
import { cleanup, render } from '@testing-library/react';

import ThemeProvider from '../../../../utils/ThemeProviderTest';
import Expense from '../expense';
import 'jest-styled-components';

describe('Expense component', () => {
    afterEach(() => {
        jest.clearAllMocks();
        cleanup();
    });

    it('tests basic snapshot', () => {
        const container = render(
            <ThemeProvider>
                <Expense />
            </ThemeProvider>,
        );
        expect(container.asFragment()).toMatchSnapshot();
    });
});
