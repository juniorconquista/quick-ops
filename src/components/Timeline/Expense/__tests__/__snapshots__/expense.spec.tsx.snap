// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Expense component tests basic snapshot 1`] = `
<DocumentFragment>
  .c7 {
  color: #f8f9fa;
  font-size: 2rem;
  font-weight: bold;
}

.c12 {
  color: #f8f9fa;
  font-size: 2rem;
  font-size: 11px;
  font-weight: bold;
  margin-bottom: 10px;
}

.c4 {
  color: #f8f9fa;
  font-size: 1.5rem;
  font-size: 13px;
}

.c8 {
  color: #f8f9fa;
  font-size: 1.5rem;
  color: #FFFFFF;
  font-weight: bold;
}

.c9 {
  color: #f8f9fa;
  font-size: 1rem;
  font-size: 14px;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #25262a;
  margin-bottom: 24px;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  width: 100%;
  min-height: 154px;
}

.c0:focus {
  outline: none;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  width: 70%;
}

.c1:focus {
  outline: none;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.c2:focus {
  outline: none;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
}

.c5:focus {
  outline: none;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.c6:focus {
  outline: none;
}

.c10 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.c10:focus {
  outline: none;
}

.c11 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.c11:focus {
  outline: none;
}

.c13 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c13:focus {
  outline: none;
}

.c3 {
  height: 59px;
  width: 59px;
  border-radius: 50%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  background-color: #f0f3f7;
  margin-bottom: 10px;
}

.c3 svg {
  height: 25px;
  width: 25px;
  fill: #6b7480;
}

@media screen and (min-width:0rem) {
  .c7 {
    font-size: 14px;
  }
}

@media screen and (min-width:37.5rem) {
  .c7 {
    font-size: 11px;
  }
}

@media screen and (min-width:62rem) {
  .c7 {
    font-size: 11px;
  }
}

@media screen and (min-width:0rem) {
  .c7 {
    margin-bottom: 0px;
    margin-right: 5px;
  }
}

@media screen and (min-width:37.5rem) {
  .c7 {
    margin-bottom: 10px;
    margin-right: 0px;
  }
}

@media screen and (min-width:62rem) {
  .c7 {
    margin-bottom: 10px;
    margin-right: 0px;
  }
}

@media screen and (min-width:0rem) {
  .c4 {
    margin-left: 5px;
  }
}

@media screen and (min-width:37.5rem) {
  .c4 {
    margin-left: 0px;
  }
}

@media screen and (min-width:62rem) {
  .c4 {
    margin-left: 0px;
  }
}

@media screen and (min-width:0rem) {
  .c8 {
    font-size: 14px;
  }
}

@media screen and (min-width:37.5rem) {
  .c8 {
    font-size: 20x;
  }
}

@media screen and (min-width:62rem) {
  .c8 {
    font-size: 20x;
  }
}

@media screen and (min-width:0rem) {
  .c8 {
    margin-bottom: 0px;
    margin-right: 5px;
  }
}

@media screen and (min-width:37.5rem) {
  .c8 {
    margin-bottom: 10px;
    margin-right: 0px;
  }
}

@media screen and (min-width:62rem) {
  .c8 {
    margin-bottom: 10px;
    margin-right: 0px;
  }
}

@media screen and (min-width:0rem) {
  .c0 {
    padding: 20px;
  }
}

@media screen and (min-width:37.5rem) {
  .c0 {
    padding: 20px 30px;
  }
}

@media screen and (min-width:62rem) {
  .c0 {
    padding: 20px 40px;
  }
}

@media screen and (min-width:0rem) {
  .c1 {
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
    -webkit-align-items: flex-start;
    -webkit-box-align: flex-start;
    -ms-flex-align: flex-start;
    align-items: flex-start;
  }
}

@media screen and (min-width:37.5rem) {
  .c1 {
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
  }
}

@media screen and (min-width:62rem) {
  .c1 {
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
  }
}

@media (max-width:1250px) {
  .c1 {
    width: 80%;
  }
}

@media screen and (min-width:0rem) {
  .c2 {
    margin-right: 0px;
    margin-bottom: 15px;
  }
}

@media screen and (min-width:37.5rem) {
  .c2 {
    margin-right: 40px;
    margin-bottom: 15px;
  }
}

@media screen and (min-width:62rem) {
  .c2 {
    margin-right: 40px;
    margin-bottom: 0px;
  }
}

@media screen and (min-width:37.5rem) {
  .c2 {
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
  }
}

@media screen and (min-width:62rem) {
  .c2 {
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
  }
}

@media screen and (min-width:0rem) {
  .c5 {
    -webkit-box-pack: start;
    -webkit-justify-content: flex-start;
    -ms-flex-pack: start;
    justify-content: flex-start;
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
    -webkit-align-items: flex-start;
    -webkit-box-align: flex-start;
    -ms-flex-align: flex-start;
    align-items: flex-start;
  }
}

@media screen and (min-width:37.5rem) {
  .c5 {
    -webkit-box-pack: justify;
    -webkit-justify-content: space-between;
    -ms-flex-pack: justify;
    justify-content: space-between;
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
  }
}

@media screen and (min-width:62rem) {
  .c5 {
    -webkit-box-pack: justify;
    -webkit-justify-content: space-between;
    -ms-flex-pack: justify;
    justify-content: space-between;
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
  }
}

@media screen and (min-width:0rem) {
  .c6 {
    margin-right: 0px;
    margin-bottom: 10px;
  }
}

@media screen and (min-width:37.5rem) {
  .c6 {
    margin-right: 0px;
    margin-bottom: 0px;
  }
}

@media screen and (min-width:62rem) {
  .c6 {
    margin-right: 15px;
    margin-bottom: 0px;
  }
}

@media screen and (min-width:37.5rem) {
  .c6 {
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
  }
}

@media screen and (min-width:62rem) {
  .c6 {
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
  }
}

@media screen and (min-width:0rem) {
  .c10 {
    margin-bottom: 10px;
  }
}

@media screen and (min-width:37.5rem) {
  .c10 {
    margin-bottom: 0px;
  }
}

@media screen and (min-width:62rem) {
  .c10 {
    margin-bottom: 0px;
  }
}

@media screen and (min-width:37.5rem) {
  .c10 {
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
  }
}

@media screen and (min-width:62rem) {
  .c10 {
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
  }
}

@media screen and (min-width:37.5rem) {
  .c11 {
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
  }
}

@media screen and (min-width:62rem) {
  .c11 {
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
  }
}

<div
    class="c0"
    data-testid="mocked-expense-component"
    width="100%"
  >
    <div
      class="c1"
      width="70%"
    >
      <div
        class="c2"
      >
        <div
          class="c3"
        >
          <svg>
            asterisk.svg
          </svg>
        </div>
        <h2
          class="c4"
          font-size="13px"
        >
          09/11/2020
        </h2>
      </div>
      <div
        class="c5"
        width="100%"
      >
        <div
          class="c6"
        >
          <h1
            class="c7"
            font-size="[object Object]"
            font-weight="bold"
          >
            TIPO
          </h1>
          <h2
            class="c8"
            color="text.contrast"
            font-size="[object Object]"
            font-weight="bold"
          />
          <p
            class="c9"
            font-size="14px"
          />
        </div>
        <div
          class="c10"
        >
          <h1
            class="c7"
            font-size="[object Object]"
            font-weight="bold"
          >
            VALOR
          </h1>
          <h2
            class="c8"
            color="text.contrast"
            font-size="[object Object]"
            font-weight="bold"
          >
            undefined undefined
          </h2>
          <p
            class="c9"
            font-size="14px"
          >
            Valor da nota: undefined undefined
          </p>
        </div>
        <div
          class="c11"
        >
          <h1
            class="c12"
            font-size="11px"
            font-weight="bold"
          >
            STATUS
          </h1>
        </div>
      </div>
    </div>
    <div
      class="c13"
    >
      <p
        class=""
      >
         Ver nota fiscal
      </p>
    </div>
  </div>
</DocumentFragment>
`;
